loan/loan_management.py

```diff
@@ -92,7 +92,16 @@ def collect_payment(self):
 
         for loan in self.loans:
             if loan.id == loan_id:
+                if loan.user_id != self._bank_account.user_id:
+                    print("You are not authorized to make payments on this loan.")
+                    return
                 if loan.status == 'approved' and loan.balance > 0:
+                    if amount > loan.balance:
+                        print(f"Payment exceeds remaining balance of {loan.balance}")
+                        return
+                    if not self._bank_account.withdraw(amount):
```

Comment:
BankAccout does not have a withdraw method..  transactionService does

Reply:
see runtime error below:
![image](https://github.com/user-attachments/assets/55592413-3f63-477a-aa6c-819515fea81e)